### PR TITLE
adding `index_search()`

### PR DESCRIPTION
**Pull request by juchem**
adding tests for `search.h`

### CODE REVIEW

File: fatal/type/impl/slice.h [hunk 1]
```diff
@@ -607,6 +607,9 @@ template <
 struct i<List<Args...>, index_sequence<Indexes...>> {
   template <typename T>
   using of = decltype(itfind<T>(inherit<indexed<Args, Indexes>...>()));
```
Comment (yfeldblum) [inline fatal/type/impl/slice.h]: Rename this to `apply` to avoid two layers of alias?

Comment (juchem) [inline fatal/type/impl/slice.h]: makes sense

**Comment by yfeldblum** [comment]
Internally:
* Fix existing remaining uses of `::of`.

**Comment by juchem** [comment]
it's failing contbuild
I'll fix it tomorrow

File: fatal/type/slice.h [hunk 2]
```diff
@@ -70,6 +70,12 @@ using index = typename i_at::i<
   make_index_sequence<size<Container>::value>
 >;
 
+template <typename Container>
+using try_index = typename i_at::ti<
```
Comment (yfeldblum) [inline fatal/type/slice.h]: Why `typename`?

Comment (juchem) [reply fatal/type/slice.h]: Leftover

**Pull request closed without merging**